packages/backend-common/src/reading/BitbucketUrlReader.ts

```diff
@@ -150,4 +196,51 @@ export class BitbucketUrlReader implements UrlReader {
     }
     return `bitbucket{host=${host},authed=${authed}}`;
   }
+
+  private async getLastCommitShortHash(
+    gitUrl: parseGitUri.GitUrl,
+  ): Promise<String> {
+    const { name: repoName, owner, ref, protocol, resource } = gitUrl;
+
+    const isHosted = resource === 'bitbucket.org';
+
+    const branch = ref ? ref : 'master';
+    const commitsApiUrl = isHosted
+      ? `${protocol}://${this.config.apiBaseUrl}/repositories/${owner}/${repoName}/commits/${branch}`
```

Comment:
I explored the [Bitbucket API](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/commits), and this endpoint looks the most appropriate to retrieve a commit hash.